vpr/src/route/connection_router.cpp

```diff
@@ -572,10 +563,10 @@ void ConnectionRouter<Heap>::timing_driven_add_to_heap(const t_conn_cost_params&
     float best_total_cost = rr_node_route_inf_[to_node].path_cost;
     float best_back_cost = rr_node_route_inf_[to_node].backward_path_cost;
 
-    float new_total_cost = next.cost;
+    float new_total_cost = next.total_cost;
     float new_back_cost = next.backward_path_cost;
 
-    if (new_total_cost < best_total_cost && ((rcv_path_manager.is_enabled()) || (new_back_cost < best_back_cost))) {
+    if (best_back_cost > new_back_cost) { // TODO: double check the performance of expansion condition
```

Comment:
Comment this. We need to only expand this node if it is a better path.  And we need to update its rr_node_route_inf data as we put it into the heap; there may be other (previously explored) paths to this node in the heap already, but they will be pruned when we pop those heap nodes later as we'll see they have inferior costs to what is in the rr_node_route_inf data for this node.  (At least some of that should be commented here; if you think there's a better place to comment the overall approach that's OK too but it should be clearly commented. Adding a link to the FPT paper would be a good idea too I think).